app/Http/Controllers/api/StudentBootcampDetailController.php

```diff
@@ -4,24 +4,36 @@
 
 use App\Http\Controllers\Controller;
 use App\Models\Student;
+use App\Service\Student\StudentBootcampDetailService;
+use Exception;
+use Illuminate\Database\Eloquent\ModelNotFoundException;
+use Illuminate\Http\JsonResponse;
 
 class StudentBootcampDetailController extends Controller
 {
-    public function __invoke($uuid)
-    {
-        Student::where('id', $uuid)->firstOrFail();
-
-        $bootcamp_detail = [
+    private StudentBootcampDetailService $studentBootcampDetailService;
 
-            'bootcamp' => [
-                [
-                    'bootcamp_id' => '1',
-                    'bootcamp_name' => 'php Fullstack Developer',
-                    'bootcamp_end_date' => ['November 2023'],
-                ],
-            ]
-        ];
-        
-        return response()->json($bootcamp_detail);
+    public function __construct(StudentBootcampDetailService $studentBootcampDetailService)
+    {
+        $this->studentBootcampDetailService = $studentBootcampDetailService;
+    }
+    /**
+     * Handle the incoming request to get bootcamp details for a specific student.
+     *
+     * @param string $studentId The UUID of the student.
+     * @throws Exception If an error occurs while executing the function.
+     * @return JsonResponse The JSON response containing containing bootcamp details.
+     */
+    public function __invoke($studentId): JsonResponse
+    {
+        try {
+            $service = $this->studentBootcampDetailService->execute($studentId);
+            return response()->json($service);
+        } catch (ModelNotFoundException $e) {
```

Comment:
Lines 32 and 34: you may consider creating custom exceptions that are more descriptive and specific.

Reply:
I'm gonna need your help so...

Reply:
I've got it! 🎉